lib/honeycomb/client.rb

```diff
@@ -35,20 +35,27 @@ def initialize(configuration:)
       @additional_trace_options = {
         presend_hook: configuration.presend_hook,
         sample_hook: configuration.sample_hook,
+        propagation_hook: configuration.http_trace_propagation_hook,
+        parser_hook: configuration.http_trace_parser_hook,
       }
 
+      @parser_hook = @additional_trace_options[:parser_hook]
```

Comment:
This is slightly awkward but `additional_trace_options` go through a few modules before landing in `Span`. Most of the trace-involved actions seem to assume the current span to be the active `self` in scope. 

For context, Rack is currently where the first `client.start_span` is called. If we're going to parse in `Rack` as we discussed, we need this hook available to Rack. I guess we could also just call it directly in `client` though? I'll play with that